### PR TITLE
Python Database Programming-SQLite

### LINKED ISSUE #430: Python Database Programming : SQLite
### Brief Summary: 

**SQLite** is a lightweight database that can provide a relational database management system with zero-configuration, In general, it is a server-less database that you can use within almost all programming languages including Python.
In this tutorial, I will cover everything about SQLite using Python.

### Key Takeaways: 

- Installation and usage
- Setup 
- Creating tables 
- Insert/Select/Update/Delete Records

### References: 

N/A

### PR DESCRIPTION
**Pull request by ahmadmardeni1**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      

#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tool:** [Grammarly](https://grammarly.com)
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you used Repl.it for your code snippets?

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

this fixes #430

### CODE REVIEW

**Comment by hectorkambow** [comment]
👋  Good afternoon @ahmadmardeni1 
Hector here the content community manager at Section - thank you so much for your 1st article contribution. 
The article is looking great so far 🚀 
Upon running your article through our 3rd party plagiarism checker it red a few flags on a few sentences - (see attached PDF)
Please revisit the article to improve and/or properly cite and sentences that need it - for any citations please use the []() syntax to place the source within the sentence(s) being cited. 

Thank you - and feel free to reach out if you have any other questions along the way - either myself or a peer reviewer will be able to help 👍 
[Python Database Programming  SQLite #442.pdf](https://github.com/section-io/engineering-education/files/5351543/Python.Database.Programming.SQLite.442.pdf)

**Comment by ahmadmardeni1** [comment]
Thank you for your time and good afternoon 🚀 @hectorkambow 
I did the changes, please take a look at it, and I will be happy to change anything you want me to change.
but I have a question, there are some natural sentences with a red flag such as "This is just a tip of iceberg", "you are ready to go" and "you have to store" 
Should I change them too?

**Comment by hectorkambow** [comment]
> Thank you for your time and good afternoon 🚀 @hectorkambow
> I did the changes, please take a look at it, and I will be happy to change anything you want me to change.
> but I have a question, there are some natural sentences with a red flag such as "This is just a tip of iceberg", "you are ready to go" and "you have to store"
> Should I change them too?

Apologies for the delayed response - no those type of sentences are fine - that tool serves as a guide. Sometimes it marks sentences that are totally fine - but its good to go through them and add any sources that may be need as a link. 

🚀

**Comment by ahmadmardeni1** [comment]
@hectorkambow  no problem. and thank you for the edits, I will pay attention to them in my next tutorial.

**Comment by hectorkambow** [comment]
👋  Good afternoon Marcus - hope your week is going well.  @mhoulmiere 
Wanted to ask if you could give this "Python Database Programming: SQLite" article from one of our new contributors a tech review whenever you have a moment - it seems like a very helpful article.  Just wanted to make sure everything was accurate before we publish.
Let me know if you have any questions 
Thank you  🚀

File: articles/python-database-programming-sqlite/index.md [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+author: ahmad-mardeni
+date: 2020-10-15T00:00:00-14:00
+topics: [Languages]
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/python-database-programming-sqlite/hero.png
+    alt: SQLite example image Python SQLite library
+---
+SQLite is an open-source and simple database engine that allows you to create a relational database and interact with it, In general, it is very lightweight and can be used within almost all programming languages including Python.
```
Comment (mhoulmiere) [inline articles/python-database-programming-sqlite/index.md:17]: Typo: There should be a period where the comma is at the end of the first sentence. "...relational database and interact with it. In general, it is very..."

File: articles/python-database-programming-sqlite/index.md [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+print(result)
+```
+
+The result will be:
+
+```python
+[('James',), ('Ahmad',)]
+```
+
+In that example, we are searching for names in table **students** where the **ID** is less than 3, and we used **ASC** to sort the column in ascending order(from the lowest value to the highest one)
```
Comment (mhoulmiere) [inline articles/python-database-programming-sqlite/index.md:158]: Missing period at the end of the sentence.

File: articles/python-database-programming-sqlite/index.md [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+pip install db-sqlite3
+
+```
+for **Linux** users you can use:
+
+```bash
+sudo apt-get install sqlite
+```
+
+In this tutorial, we will use **DB Browser for SQLite**, you can download it from [here](https://sqlitebrowser.org/dl/)
```
Comment (mhoulmiere) [inline articles/python-database-programming-sqlite/index.md:45]: Missing the period at the end of the sentence.

File: articles/python-database-programming-sqlite/index.md [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+    alt: SQLite example image Python SQLite library
+---
+SQLite is an open-source and simple database engine that allows you to create a relational database and interact with it, In general, it is very lightweight and can be used within almost all programming languages including Python.
+<!--more-->
+
+### Why SQLite?
+- Scalability
+- Concurrency
+- Centralization
+- Control
```
Comment (mhoulmiere) [inline articles/python-database-programming-sqlite/index.md:24]: I would add any docs/references to corroborate this assertion or explain it in your own words.

File: articles/python-database-programming-sqlite/index.md [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+connection.commit()
+```
+
+Then we have to close the connection to the database by using:
+
+```python
+connection.close()
+```
+
+When the connection is closed then a ROLLBACK will happen in the next connection to the database to any outstanding transaction. (The ROLLBACK statement cancels transaction that did any changes to the database).
```
Comment (mhoulmiere) [inline articles/python-database-programming-sqlite/index.md:210]: This statement is a little confusing, you should revise to make it a bit more clear that outstanding transactions == un-committed changes. Also, the parenthesis should be removed from the ROLLBACK clarification sentence.

**Review by mhoulmiere — CHANGES_REQUESTED** [review]
Great job on this article! It is a great introduction to using Python to interact with SQLite DBs. I've noted a few typos and made one or two suggestions that you should consider. Overall, not much needed in terms of revisions. 

Two final suggestions (not requirements) would be to consider adding a prerequisite section to clarify what the reader should have familiarity with before reading your article and to add some more links to documentation that they can leverage if they need or would like more information.

**Comment by ahmadmardeni1** [comment]
Good evening and thank you for your time🚀 @mhoulmiere 
I did all of the changes you asked for.

**Pull request merged**